File: leetcode-submissions/2658_Maximum_Number_of_Fish_in_a_Grid.ts
function findMaxFish(grid: number[][]): number {
    let maxFishes:number =0;

    for(let i:number =0;i<grid.length;i++){
        for(let j:number =0;j<grid[0].length;j++){
maxFishes=Math.max(maxFishes,BFS(i,j,grid));
        }
    }
    return maxFishes;
};
function BFS (i:number ,j:number ,grid:number[][]):number {
    const m:number =grid.length;
    const n:number =grid[0].length;
    const directions :[number,number][]=[[0,1],[0,-1],[1,0],[-1,0]];
    const queue:[number,number][]=[[i,j]];
    let Fishes:number =0;
    
    while(queue.length>0){
    const [i,j]=queue.shift();
    if(grid[i][j]===0){
        continue;
    }
    Fishes+=grid[i][j];

    grid[i][j]=0;
    for(const [row,col] of directions){
        const i_:number =row+i;
        const j_:number=col+j;
    if(i_>=0 && j_>=0 && i_<m && j_<n && grid[i_][j_]>0){
        queue.push([i_,j_]);
    }
        
    }



}
return Fishes;



}

File: leetcode-submissions/2658_Maximum_Number_of_Fish_in_a_Grid.cpp
class Solution {
public:
    vector<vector<int>>directions ={{0,1},{0,-1},{1,0},{-1,0}};
    int m,n;
    int dfs(int i ,int j,vector<vector<int>>&grid){
        if(i>=m or j>=n or i<0 or j<0 or grid[i][j]==0){
            return 0;
        }
        int Fishes =grid[i][j];
        grid[i][j]=0;
        for(auto dir :directions){
            int i_=i+dir[0];
            int j_=j+dir[1];
            Fishes+=dfs(i_,j_,grid);

        }
        return Fishes;


    }
    
    int findMaxFish(vector<vector<int>>& grid) {
        int maxFishes =INT_MIN;
        m=grid.size();
        n=grid[0].size();

        for(int i=0;i<grid.size();i++){
            for(int j=0;j<grid[0].size();j++)
            {
                if(grid[i][j]>0){
                    maxFishes=max(maxFishes, dfs(i,j,grid));
                }
            }
        }
        return maxFishes;
    }
};